include/drivers/st25r3911b/nfca.h

```diff
@@ -0,0 +1,308 @@
+ *           procedure, @ref nfca_cb.anticollision_completed
+ *           is called.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_anticollision_start(void);
+
+/** @brief Exchange the specified amount of data.
+ *
+ *  @details This is a generic function for NFC-A data exchange.
+ *
+ *  @param[in] tx TX data buffer.
+ *  @param[in,out] rx RX data buffer.
+ *  @param[in] fdt Maximum frame delay time (according to NFC Forum Digital 2.0
+ *                 6.10.1).
+ *  @param[in] auto_crc If set, the CRC is generated automatically by the
+ *                      NFC Reader. Otherwise, the CRC must be added
+ *                      manually to the TX buffer.
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_transfer(struct nfca_buf *tx, struct nfca_buf *rx,
+			     u32_t fdt, bool auto_crc);
+
+/** @brief Exchange the specified amount of data with an automatically
+ *         generated CRC.
+ *
+ *  @param[in] tx TX data buffer.
+ *  @param[in,out] rx RX data buffer.
+ *  @param[in] fdt Maximum frame delay time (according to NFC Forum Digital 2.0
+ *                 6.10.1).
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+#define st25r3911b_nfca_transfer_with_crc(tx, rx, fdt) \
+	st25r3911b_nfca_transfer(tx, rx, fdt, true)
+
+/** @brief Exchange the specified amount of data with a manually
+ *         generated CRC.
+ *
+ *  @details The CRC must be added manually to the TX buffer.
+ *
+ *  @param[in] tx TX data buffer.
+ *  @param[in,out] rx RX data buffer.
+ *  @param[in] fdt Maximum frame delay time (according to NFC Forum Digital 2.0
+ *                 6.10.1).
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+#define st25r3911b_nfca_transfer_without_crc(tx, rx, fdt) \
+	st25r3911b_nfca_transfer(tx, rx, fdt, false)
+
+/** @brief Send NFC-A tag sleep command.
+ *
+ *  @details This function sends the tag sleep command
+ *           (according to NFC Forum Digital 2.0 6.9).
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_tag_sleep(void);
+
+/** @brief Calculate CRC 16 for NFC-A payload.
+ *
+ *  @details This function generates an NFC-A CRC (according to
+ *           NFC Forum Digital 2.0 6.4.1.3).
+ *
+ *  @param[in] data Payload data.
+ *  @param[in] len Data length.
+ *  @param[out] crc_val NFC-A CRC value.
+ */
+void st25r3911b_nfca_crc_calculate(u8_t *data, size_t len,
+				   struct nfca_crc *crc_val);
+
+/** @brief Process NFC-A library.
+ *
+ *  @details This function must be called periodically from
+ *           thread or other context to process the NFC reader state
```

Comment:
> or other context

considering that you may sleep in this code this cannot be any other context then tread

Reply:
ping